File: python/caesar_cipher.py
import re

def caesar_cipher(string, shift_amount):
    string = list(string)
    let = "^[a-zA-Z]"
    ans = []

    for i,x in enumerate(string):
        if re.match(let,x):
            j = 97 if x.islower() else 65
            temp = ((ord(x)-j+shift_amount)%26)+j
            x = chr(temp)
        ans.append(x)

    return ''.join(ans)

File: js/caesarCipher.js


exports.caesarCipher = function(str, num) {
    str = str.split('')
    re = /^[a-zA-Z]/
    ans = []
    let j, temp;

    for(let i = 0; i<str.length; i++){
        if(re.test(str[i])){
            j = str[i]===str[i].toUpperCase() ? 65 : 97
            temp = (((str[i].charCodeAt(0))-j+num)%26)
            temp = temp<0 ? temp+=(j+26) : temp+=j
            ans.push(String.fromCharCode(temp))
        }
        else{
            ans.push(str[i])
        }

    }
    return ans.join('')
}

